packages/ng/tooltip/trigger/tooltip-trigger.directive.ts

```diff
@@ -1,156 +1,339 @@
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
 import { SafeHtml } from '@angular/platform-browser';
-import { ALuPopoverTrigger, LuPopoverPosition, LuPopoverScrollStrategy, LuPopoverTarget } from '@lucca-front/ng/popover';
-import { LuTooltipPanelComponent } from '../panel/tooltip-panel.component';
+import { LuPopoverPosition } from '@lucca-front/ng/popover';
+import { BehaviorSubject, Observable, Subject, combineLatest, merge, switchMap, timer } from 'rxjs';
+import { debounce, debounceTime, filter, map } from 'rxjs/operators';
+import { LuTooltipPanelComponent } from '../panel';
+
+let nextId = 0;
 
 @Directive({
 	selector: '[luTooltip]',
 	standalone: true,
 })
-export class LuTooltipTriggerDirective extends ALuPopoverTrigger<LuTooltipPanelComponent, LuPopoverTarget> implements AfterViewInit, OnDestroy {
-	@Input('luTooltip') set tooltipContent(c: string | SafeHtml) {
-		if (this.panel) {
-			this.panel.content = c;
-		}
+export class LuTooltipTriggerDirective implements AfterContentInit {
+	#overlay = inject(Overlay);
 
-		this._tooltipContent = c;
-	}
-	/** when trigger = hover, delay before the popover panel appears, default 300ms */
-	@Input('luTooltipEnterDelay') set inputEnterDelay(d: number) {
-		this.enterDelay = d;
+	#host = inject<ElementRef<HTMLElement>>(ElementRef);
+
+	#renderer = inject(Renderer2);
+
+	#destroyRef = inject(DestroyRef);
+
+	#cdr = inject(ChangeDetectorRef);
+
+	@Input()
+	luTooltip: string | SafeHtml;
+
+	#openDelay$ = new BehaviorSubject<number>(300);
+
+	@Input({ transform: numberAttribute })
+	set luTooltipEnterDelay(delay: number) {
+		this.#openDelay$.next(delay);
 	}
-	/** when trigger = hover, delay before the popover panel disappears, default 100ms */
-	@Input('luTooltipLeaveDelay') set inputLeaveDelay(d: number) {
-		this.leaveDelay = d;
+
+	#closeDelay$ = new BehaviorSubject<number>(100);
+
+	@Input({ transform: numberAttribute })
+	set luTooltipLeaveDelay(delay: number) {
+		this.#closeDelay$.next(delay);
 	}
-	/** disable popover apparition */
-	@Input('luTooltipDisabled') set inputDisabled(d: boolean) {
-		this.disabled = d;
-		if (this._handleTabindex) {
-			this._setTabindex(d ? null : 0);
+
+	#disabled = false;
+
+	@Input({ transform: numberAttribute })
+	set luTooltipDisabled(disabled: boolean) {
+		this.#disabled = disabled;
+		if (disabled) {
+			this.setAccessibilityProperties(null);
 		}
 	}
 
-	@Input('luTooltipPosition') set inputPosition(pos: LuPopoverPosition) {
-		this.target.position = pos;
-	}
+	@Input()
+	luTooltipPosition: LuPopoverPosition = 'above';
 
-	@Input('luTooltipWhenEllipsis') public set inputWhenEllipsis(we: boolean) {
-		this.whenEllipsis = we;
-	}
+	@Input({ transform: booleanAttribute })
+	luTooltipWhenEllipsis = false;
+
+	resize$ = new Observable((observer) => {
+		new ResizeObserver(() => {
+			observer.next();
+		}).observe(this.#host.nativeElement);
```

Comment:
We could "unobserve" in the teardown logic of this observable: 

```ts
new Observable(observer => {
	// Observe here

	// Teardown logic
	return () => {
		// Unobserve here
	}
})

```

Reply:
I'm not sure I understand the goal you want to achieve with this?

Reply:
Nvm, got it.